backend/kernelCI_app/management/commands/helpers/kcidbng_ingester.py

```diff
@@ -1,73 +1,56 @@
-import kcidb
-import tempfile
-import os
-import argparse
+from concurrent.futures import ThreadPoolExecutor, as_completed
+import gzip
+import hashlib
 import json
-import time
 import logging
-import yaml
-from concurrent.futures import ThreadPoolExecutor, as_completed
+import os
+from queue import Queue, Empty
 import requests
-import hashlib
+import tempfile
 import threading
-from queue import Queue
-import queue
+import time
 import traceback
-import gzip
+from typing import Any, Literal, Optional
+import yaml
+import kcidb_io
+
+from kernelCI_app.management.commands.helpers.process_submissions import (
+    insert_submission_data,
+)
 
-# default database
-DATABASE = "postgresql:dbname=kcidb user=kcidb password=kcidb host=localhost port=5432"
 VERBOSE = 0
-STORAGE_TOKEN = os.environ.get("STORAGE_TOKEN", None)
-STORAGE_BASE_URL = os.environ.get("STORAGE_BASE_URL", None)
 LOGEXCERPT_THRESHOLD = 256  # 256 bytes threshold for logexcerpt
 CONVERT_LOG_EXCERPT = False  # If True, convert log_excerpt to output_files url
+
 CACHE_LOGS = {}
+CACHE_LOGS_SIZE_LIMIT = 100000  # Arbitrary limit for cache_logs size, adjust as needed
 cache_logs_lock = threading.Lock()
 
+STORAGE_TOKEN = os.environ.get("STORAGE_TOKEN", None)
+STORAGE_BASE_URL = "https://files-staging.kernelci.org"
```

Comment:
To avoid having to patch things in the server:

```suggestion
STORAGE_BASE_URL = os.environ.get("STORAGE_BASE_URL", "https://files-staging.kernelci.org")
```

But I would suggest moving most of these constants to settings in a future PR.

Reply:
As matter of fact, this made me notice that Denys got a few PRs merged since you copied his code from https://github.com/kernelci/kcidb-ng 

Make sure to be synced with [his latest closed PRs](https://github.com/kernelci/kcidb-ng/pulls?q=is%3Apr+is%3Aclosed) before merging it here!